src/main/java/org/apache/mesos/hdfs/HDFSScheduler.java

```diff
@@ -81,11 +100,11 @@ public void offerRescinded(SchedulerDriver driver, OfferID offerId) {
   public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterInfo masterInfo) {
     try {
       persistentState.setFrameworkId(frameworkId);
+      log.info("Registered framework frameworkId=" + frameworkId.getValue());
+
     } catch (InterruptedException | ExecutionException e) {
       log.error("Error setting framework id in persistent state", e);
-      throw new RuntimeException(e);
```

Comment:
there are multiple issues here... 
1) Using RuntimeException is a bad idea... throwing something that extends RuntimeException would be preferred.    
2) There is nothing to handle the throw.  Except main().
3) It appears to me that the thread for ConfigServer isn't a daemon thread, so the runtime exception will be handled by main, but will not exit the process as you desire.

I'll add a runtime exception and a main() handler to exit.